ISSITES-5787 Top Level Public Sites Nav changes

liaoethan opened the pull request.

liaoethan committed:
ISSITES-5787 add custom fields for navigation menu items

liaoethan committed:
ISSITES-5787 create primary navigation adt for site initializer

liaoethan committed:
ISSITES-5787 move over top level primary nav markup and styles from liferay.com and refactor

liaoethan committed:
ISSITES-5787 refactor styles

liaoethan committed:
ISSITES-5787 add dark mode styling for top level nav items

liaoethan committed:
ISSITES-5787 refactor localized expando values method in nav adt

liaoethan committed:
ISSITES-5787 add color-neutral-2/color-neutral-8 stylebook token for learn site initializer

ISSITES-5787 add color-neutral-8 stylebook token for learn site initializer

liaoethan committed:
ISSITES-5787 refactor more adt styling

jotchon commented:
Just started reviewing :)

jotchon commented on modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/fragments/group/public-sites-navigation/fragments/public-sites-navigation/index.scss:
```diff
@@ -39,26 +39,28 @@ nav.public-sites-navigation {
-						.adt-nav-caret-bottom-icon {
-							color: var(--white);
+						.adt-nav-text {
+							.adt-nav-title {
```
If there are no styles on the parent CSS selector (i.e. .adt-nav-text), ideally we would just keep it on the same line like .adt-nav-text .adt-nav-title instead of nesting.

jotchon commented on modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/ddm-templates/navigation-utilities/ddm-template.ftl:
```diff
@@ -64,14 +64,10 @@
 											<div class="adt-submenu-item-preheader color-neutral-3 font-weight-semi-bold">${preheaderText}</div>
 										</#if>
 
-										<#if stringUtil.equals(menuItemType, "Image")>
```
Are we removing the ability to add images to the navigation?

liaoethan replied:
For this section that was removed, it was for the heading change if the menu item type was an image
```
<#if stringUtil.equals(menuItemType, "Image")>
	<div class="adt-submenu-item-title font-size-paragraph-small font-weight-semi-bold">${navTertiaryItem.getName()}</div>
<#else>
	<div class="adt-submenu-item-title font-size-paragraph-base font-weight-semi-bold">${navTertiaryItem.getName()}</div>
</#if>
```

This doesn't apply to the new navigation. The images are still there in L58-60:
```
	<#if stringUtil.equals(menuItemType, "Image") && imageURL?has_content>
		<img class="adt-submenu-item-image" loading="lazy" src="${imageURL}" />
	</#if>
```

liaoethan replied:
But this doesn't apply to the new navigation

jotchon commented on modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/style-books/learn/frontend-tokens-values.json:
```diff
@@ -42,6 +42,10 @@
 		"value": "#00256C"
 	},
+	"colorNeutral2": {
+		"cssVariableMapping": "color-neutral-2",
```
You could squash the two commits that add color to frontend tokens.  No need for them to be separate commits.

jotchon commented on modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/fragments/group/public-sites-navigation/fragments/public-sites-navigation/index.scss:
```diff
@@ -389,53 +402,79 @@ html body:not(.has-edit-mode-menu) .content #main-content.layout-content {
-			transform: translate3d(0px, 0px, 0px) scale3d(1, 1, 1)
-				rotateX(180deg) rotateY(0deg) rotateZ(0deg)
-				skew(0deg, 0deg);
+		.adt-nav-caret-bottom-icon {
```
Is the rotation not possible anymore?

jotchon commented on modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/ddm-templates/navigation-utilities/ddm-template.ftl:
```diff
@@ -1,14 +1,16 @@
+					<span class="adt-nav-caret-bottom-icon align-self-center">
+						<svg class="lexicon-icon lexicon-icon-caret-bottom" role="presentation" viewBox="0 0 512 512"><use xlink:href="/o/admin-theme/images/clay/icons.svg#caret-bottom"></use></svg>
+					</span>
+					<span class="adt-nav-caret-top-icon align-self-center">
```
Is there a reason for adding more markup?  Previously the caret was rotating.

jotchon reviewed (commented):
It looks like the Public Site Nav is coming along well!  There were a few things that were not working when I was testing:
- SiteNavigationMenuItem configurations are not working, specifically the dropdowns
    - Dropdown don’t have any selections
    - Image custom field is missing
- Up/Down Caret:  Could we not achieve this like we did previously with the rotation?
- Menu does not open, should this be working?
- Nav dropdowns should be right aligned, but they are currently centered

This is not apart of the ticket, but Abel mentioned awhile ago that we need to change the logo.  It should just be logo and subsite name, not logo, Liferay, and subsite name.  Do you mind adding that to this ticket?  It should be simple.
<img width="584" alt="Screen Shot 2022-11-02 at 5 36 58 AM" src="https://user-images.githubusercontent.com/35710155/199491782-9f02efe7-6ee5-4bd7-9ad3-cf21488158b7.png">

Let me know if you have any questions about my comments above.

jotchon commented:
More notes:
- The box shadow is also missing from the mega dropdown menu.
- Note to use "dropdown-open" since the dropdown provider is not available yet
- Expando Value Permissions need to be viewable for guests

liaoethan committed:
ISSITES-5787 refactor css and remove top caret

ISSITES-5787 caret changes

liaoethan commented:
Hi @jotchon, I made the following changes:

- Squashed redundant commits (stylebook tokens, etc.)
- Removed the top caret
- Make refactoring changes with scss
- Added the box shadow
- Right aligned menu items
- Removal of liferay text from liferay logo

For the image configuration change, I added a comment. Let me know if you want me to make any additional changes. Thanks for all of the feedback!

jotchon commented:
Just started reviewing :)

liaoethan committed:
ISSITES-5787 remove liferay text from liferay logo

liaoethan committed:
ISSITES-5787 right align menu items

jotchon commented:
Hi @liaoethan, there are few things that could be improved with this PR:

- it could be more organized.  There are a lot of commits and it is difficult to review.
- Many of the commits say "refactor".  This could be more descriptive.
- The squashed commits have two lined messages (i.e. color-neutral commit).  I believe you forgot to comment out the other commit message.
- since this is the first commit for the ADT, you could combine those changes into one commit.  You might keep those changes separate for yourself, but we don't need to see all the changes for this.  

Not sure if this is clear, but let me know if you have questions about this.  Otherwise, this is great work!

liaoethan commented:
Hi @jotchon, thanks for the feedback. I'll create another PR and make the commits more concise.

liaoethan commented:
I created a condensed pull here: https://github.com/jotchon/liferay-portal/pull/4

I see extra commits but is that being caused by an outdated master?
